crates/sui-network/src/state_sync/tests.rs

```diff
@@ -256,6 +263,181 @@ async fn isolated_sync_job() {
+        .build();
+    let network_1 = build_network(|router| router.add_rpc_service(server));
+    let (event_loop_1, _handle_1) = builder.build(network_1.clone());
+    let (builder, server) = Builder::new().store(SharedInMemoryStore::default()).build();
+    let network_2 = build_network(|router| router.add_rpc_service(server));
+    let (event_loop_2, _handle_2) = builder.build(network_2.clone());
+    network_1.connect(network_2.local_addr()).await.unwrap();
+
+    // Init the root committee in both nodes
+    event_loop_1.store.inner_mut().insert_genesis_state(
+        ordered_checkpoints.first().cloned().unwrap(),
+        empty_contents(),
+        committee.committee().to_owned(),
+    );
+    event_loop_2.store.inner_mut().insert_genesis_state(
+        ordered_checkpoints.first().cloned().unwrap(),
+        empty_contents(),
+        committee.committee().to_owned(),
+    );
+
+    // Node 2 will have all the data at first
+    {
+        let mut store = event_loop_2.store.inner_mut();
+        for checkpoint in ordered_checkpoints.clone() {
+            store.insert_checkpoint(&checkpoint);
+            store.insert_checkpoint_contents(&checkpoint, empty_contents());
+            store.update_highest_synced_checkpoint(&checkpoint);
+        }
+    }
+    // Prune first 10 checkpoint contents from Node 2
+    {
+        let mut store = event_loop_2.store.inner_mut();
+        for checkpoint in &ordered_checkpoints[0..(oldest_checkpoint_to_keep as usize)] {
+            store.delete_checkpoint_content_test_only(checkpoint.sequence_number)?;
+        }
+        // Now Node 2 has deleted checkpoint contents from range [0, 10) on local store
+        assert_eq!(
+            store.get_lowest_available_checkpoint(),
+            oldest_checkpoint_to_keep
+        );
+        assert_eq!(
+            store
+                .get_highest_synced_checkpoint()
+                .unwrap()
+                .sequence_number,
+            ordered_checkpoints.last().unwrap().sequence_number
+        );
+        assert_eq!(
+            store
+                .get_highest_verified_checkpoint()
+                .unwrap()
+                .sequence_number,
+            ordered_checkpoints.last().unwrap().sequence_number
+        );
+    }
+
+    // Node 1 will know that Node 2 has the data starting checkpoint 10
+    event_loop_1.peer_heights.write().unwrap().peers.insert(
+        network_2.peer_id(),
+        PeerStateSyncInfo {
+            genesis_checkpoint_digest: *ordered_checkpoints[0].digest(),
+            on_same_chain_as_us: true,
+            height: *ordered_checkpoints.last().unwrap().sequence_number(),
+            lowest: oldest_checkpoint_to_keep,
+        },
+    );
+
+    // Get handle to node 1 store
+    let store_1 = event_loop_1.store.clone();
+
+    // Sync the data
+    // Start both event loops
+    tokio::spawn(event_loop_1.start());
+    tokio::spawn(event_loop_2.start());
+
+    loop {
```

Comment:
we should have a timeout here to avoid a failed test taking too long to finish

Reply:
ah i see, sure i will add a meaningful timeout